client/spec/src/main/asciidoc/client.asciidoc

```diff
@@ -17,6 +17,392 @@
+
+== Focus of release 1.1
+
+A first version of the client API is planned with MicroProfile GraphQL 1.1. For this first step, we propose to focus on the following core features.
+
+==== GraphQL support
+
+* Operations: queries and mutations
+* Partial results
+* Errors
+* Arguments
+* Variables
+* Interface
+
+==== HTTP supports
+
+* Headers: reading and writing
+* Connection timeout
+* Read timeout
+
+==== JSON-B support
+
+* Format directives on fields: dates...
+
+==== MicroProfile Config
+
+Configuration exclusively based on MicroProfile Config.
+
+==== Synchronous client API
+
+We propose a synchronous-only API for this first release. 
+
+A reactive API, similar to JAX-RS client API (see chapter 5.7) will be studied for a next release.
+
+==== Java 8 support
+
+Java 8 is still widely used in the industry and we propose to stick to it for a broader adoption.
+
+== Next steps
+
+To be studied for next releases:
+
+* GraphQL subscription, union
+* Reactive API
+* https://spec.graphql.org/draft/#sec-Document[Multiple operations inside a GraphQL document]
+
+== API building-blocks
+
+All classes described in this document are part of the package __org.eclipse.microprofile.graphql.client__.
+
+The API is based on 5 main classes:
+
+* **Builder**: prepares and builds GraphQL operations
+* **Request**: represents a read-to-run GraphQL operation
+* **Client**: executes operations against a GraphQL server endpoint, reusable and thread-safe 
+* **Response**: a holder for the GraphQL response including both data (which can be partial) and GraphQL errors if any
+* **InputType**: enables to build complex input objects.
+
+== Code examples
+
+=== Building a basic query
+
+In this example, we want to get the name of all heroes. The GraphQL query that fits that needs is:
+[source,json]
+----
+allHeroes {
+	name
+}
+----
+The way to build this query with the API is:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.QUERY)
+  .addRootField(new Field("allHeroes", 
+    new ArrayList<>(asList(new Field("name")))));
+----
+To be more concise and expressive, we also propose a lighter syntax based on predefined static methods:
+[source,java]
+----
+Builder builderWithSugar = new Builder(Operation.Type.QUERY)
+       .addRootField("allHeroes", fields(field("name")));
+----
+
+The static methods to obtain this lighter syntax are described in the <<appendix_static_methods>>.
+
+=== Building a more complex query
+
+In this example, we want to get all heroes from a given city, including the name of the teams they belong to.
+The GraphQL query that fits this need is:
+
+[source,json]
+----
+query {
+  allHeroesIn(city: "New York, NY") {
+	name
+	currentLocation
+	teamAffiliations {
+  		name
+	}
+  }
+----
+The way to build this query with the API is:
+[source,java]
+----
+Builder builder = new Builder(Operation.Type.QUERY)
+       .addRootField(new Field("allHeroesIn",
+               new HashSet<Argument>(asList(
```

Comment:
Why is this a `Set`? It might be a `Map`, so the argument names can only be used only once; but a `Set` feels strange to me.